baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/DeprecatedConjure.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.MemberReferenceTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.Tree;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        severity = BugPattern.SeverityLevel.ERROR,
+        summary = "You should not use Conjure endpoints that are marked for removal as that may block"
+                + " upgrades in the near future. You can explicitly disable this check on a case-by-case basis using"
+                + " @SuppressWarnings(\"DeprecatedConjure\").")
+public final class DeprecatedConjure extends BugChecker
```

Comment:
Naming is hard. "Deprecated conjure" may seem like conjure is deprecated, and doesn't make it obvious that deprecated-for-removal is more severe than standard deprecation.

While a bit wordy, perhaps something along the lines of `ConjureEndpointDeprecatedForRemoval`?

Reply:
Mostly agree, though still unresolved what we should do if client code is using a deprecated-for-removal object or field - makes having the term "EndPoint" in the name a little confusing.  Not a huge deal, or we may choose to make that a separate check if we do implement it.

Reply:
>  if client code is using a deprecated-for-removal object or field

Only endpoints support this flag, objects and fields aren't deprecated-for-removal